### ISSUE TITLE
create_chat_completion is stuck in versions 0.2.84 and 0.2.85 for Mac Silicon

### ISSUE THREAD

**Issue by mobeetle** [abetlen/llama-cpp-python#1648]
# Prerequisites

Version 0.2.84 or 0.2.85 and using create_chat_completion method.
Tried different GGUF models.

Please answer the following questions for yourself before submitting an issue.

- [ X ] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [ X ] I carefully followed the [README.md](https://github.com/abetlen/llama-cpp-python/blob/main/README.md).
- [ X ] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [ X ] I reviewed the [Discussions](https://github.com/abetlen/llama-cpp-python/discussions), and have a new bug or useful enhancement to share.

# Expected Behavior

Provide a result as described in the documentation. 

# Current Behavior

Inference is stuck (I let it run for 5 minutes).
After downgrading to version 0.2.83 everything runs without a single change in the code.

# Environment and Context

Mac M1 MAX, 32GB RAM, MacOS 14.5, Python 3.12, llama-cpp-python 0.2.84/5.

**Comment by shamitv** [comment]
Can't recreate this issue with :

M1 , Python 3.12 , 0.2.85 , Phi-3 and Llama 3.1 

![image](https://github.com/user-attachments/assets/5ae9ca01-5bd9-4881-912b-5c75fe1aa6c8)

![image](https://github.com/user-attachments/assets/9fd7792c-bf66-4a5d-80f4-64e77fab2c57)

**Comment by mobeetle** [comment]
Hi, 

thanks for checking.

When I recreate your test, it is working.

The problem seems to be when using JSON schema with 0.2.84/5. Working with 0.2.83. 

Please find attached Jupyter NB. It locks on Mac.

Kind regards

Lukáš

￼



On 3. 8. 2024, at 11:52, Shamit Verma ***@***.***> wrote:


Can't recreate this issue with :

M1 , Python 3.12 , 0.2.85 , Phi-3 and Llama 3.1

image.png (view on web) <https://github.com/user-attachments/assets/5ae9ca01-5bd9-4881-912b-5c75fe1aa6c8>
image.png (view on web) <https://github.com/user-attachments/assets/9fd7792c-bf66-4a5d-80f4-64e77fab2c57>

**Comment by shamitv** [comment]
Don't see the attachment somehow.

**Comment by handshape** [comment]
Yup - willing to bet this is fixed in https://github.com/abetlen/llama-cpp-python/pull/1649 - there's a whole cluster of issues that will get cleared with this change.

**Comment by shamitv** [comment]
Version 0.2.87 does not have this issue 

![image](https://github.com/user-attachments/assets/785427d8-3031-40e4-b965-0985b65eb2bf)
